--- meson.build.orig	2020-11-30 10:37:31.535903600 -0800
+++ meson.build.new	2020-11-30 11:01:24.308909000 -0800
@@ -58,6 +58,10 @@
   dependency('libavutil', version: '>= 56.14.100'),
 ]
 
+if host_system == 'windows'  and cc.get_id() == 'msvc'
+  lib_deps += dependency('libpthreads')
+endif
+
 vaapi_dep = dependency('libva', version: '>= 1.1.0', required: get_option('vaapi'))
 if vaapi_dep.found()
   lib_deps += vaapi_dep
@@ -95,6 +99,14 @@
   lib_src += files('decoder_vt.c')
 endif
 
+lib_name_prefix = []
+lib_name_suffix = []
+if host_system == 'windows'
+  # Follow Windows library convention
+  lib_name_prefix = ''
+  lib_name_suffix = 'lib'
+endif
+
 libsxplayer = library(
   'sxplayer',
   lib_src,
@@ -103,6 +115,8 @@
   install_rpath: install_rpath,
   version: meson.project_version(),
   link_args: lib_link_args,
+  name_prefix: lib_name_prefix,
+  name_suffix: lib_name_suffix
 )
 
 lib_header = configure_file(
