diff -ruN tmp/sxplayer-9.5.1/CMakeLists.txt sxplayer-9.5.1/CMakeLists.txt
--- tmp/sxplayer-9.5.1/CMakeLists.txt	1969-12-31 16:00:00.000000000 -0800
+++ sxplayer-9.5.1/CMakeLists.txt	2020-09-21 12:58:43.396933959 -0700
@@ -0,0 +1,43 @@
+cmake_minimum_required(VERSION 3.14)
+project (sxplayer)
+
+file (GLOB_RECURSE SXPLAYER_SRC_FILES *.c *.h)
+if (NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+list (FILTER SXPLAYER_SRC_FILES EXCLUDE REGEX "decoder_vt.c")
+endif()
+list (FILTER SXPLAYER_SRC_FILES EXCLUDE REGEX "test-prog.c")
+
+if (${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+set(SXPLAYER_BUILD_TYPE STATIC)
+else()
+set(SXPLAYER_BUILD_TYPE SHARED)
+endif()
+
+add_library(sxplayer ${SXPLAYER_BUILD_TYPE} ${SXPLAYER_SRC_FILES})
+target_include_directories(sxplayer PUBLIC 
+        ${PTHREADS_INCLUDE_DIRS}
+        ${FFMPEG_INCLUDE_DIRS}
+        ${GLEW_INCLUDE_DIRS}
+        ${GLFW_INCLUDE_DIRS}
+)
+set_target_properties(sxplayer PROPERTIES COMPILE_FLAGS
+    "${PTHREADS_CFLAGS} ${FFMPEG_CFLAGS} ${GLEW_CFLAGS} ${GLFW_CFLAGS}"
+)
+target_link_libraries(sxplayer PUBLIC 
+        ${PTHREADS_LIBRARIES}
+        ${FFMPEG_LIBRARIES}
+        ${GLEW_LIBRARIES}
+        ${GLFW_LIBRARIES}
+)
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    target_link_libraries(sxplayer PUBLIC
+        "-framework CoreFoundation -framework VideoToolbox -framework CoreMedia -framework QuartzCore"
+        "-framework OpenGL"
+    )
+endif()
+target_link_directories(sxplayer PUBLIC 
+        ${PTHREADS_LIBRARY_DIRS}
+        ${FFMPEG_LIBRARY_DIRS}
+        ${GLEW_LIBRARY_DIRS}
+        ${GLFW_LIBRARY_DIRS}
+)
